trackfilter.cc

```diff
@@ -57,90 +72,54 @@ int TrackFilter::trackfilter_opt_count()
   return res;
 }
 
-int TrackFilter::trackfilter_parse_time_opt(const char* arg)
+qint64 TrackFilter::trackfilter_parse_time_opt(const char* arg)
 {
-  int sign = 1;
-  const char* cin = arg;
-  char c;
-
-  time_t t0 = 0;
-  time_t t1 = 0;
-
-  while ((c = *cin++)) {
-    time_t seconds;
-
-    if (c >= '0' && c <= '9') {
-      t1 = (t1 * 10) + (c - '0');
-      continue;
+  qint64 result;
+
+  QRegularExpression re("^([+-]?\\d+)([dhms])$", QRegularExpression::CaseInsensitiveOption);
+  assert(re.isValid());
+  QRegularExpressionMatch match = re.match(arg);
+  if (match.hasMatch()) {
+    bool ok;
+    result = match.captured(1).toLong(&ok);
+    if (!ok) {
+      fatal(MYNAME "-time: invalid quantity in move option \"%s\"!\n", qPrintable(match.captured(1)));
```

Comment:
Optional throughout: Use Fatal() and Warning() to avoid qPrintable
Somewhere I have a patch that allows you to create a Logger object that accepts MYNAME ...

Reply:
Maybe later.  I think you already commited this in src/core/logging.h, e.g.
stmsdf.cc:      Warning() << MYNAME << "Invalid internal field type" << what;